src/components/sbb-tag-group/sbb-tag-group.e2e.ts

```diff
@@ -1,588 +1,558 @@
-        expect(changeSpy).toHaveReceivedEventTimes(1);
-        expect(await element.getProperty('value')).toEqual([]);
+        expect(changeSpy.count).to.be.equal(1);
+        expect(element.value).to.be.empty;
       });
 
       it('should not emit events by setting checked programmatically [prop]', async () => {
-        const changeSpy = await page.spyOnEvent('change');
-        const inputSpy = await page.spyOnEvent('input');
-        const tag2 = await page.find('sbb-tag#sbb-tag-2');
+        const changeSpy = new EventSpy('change');
+        const inputSpy = new EventSpy('input');
+        const tag2 = document.querySelector('sbb-tag#sbb-tag-2') as SbbTag;
 
-        await tag2.setProperty('checked', false);
-        await page.waitForChanges();
+        tag2.checked = false;
+        await element.updateComplete;
 
-        expect(tag2).not.toHaveAttribute('checked');
-        expect(await tag2.getProperty('checked')).toBe(false);
-        expect(inputSpy).not.toHaveReceivedEvent();
-        expect(changeSpy).not.toHaveReceivedEvent();
+        expect(tag2).not.to.have.attribute('checked');
+        expect(tag2.checked).to.be.equal(false);
+        expect(inputSpy.count).not.to.be.greaterThan(0);
+        expect(changeSpy.count).not.to.be.greaterThan(0);
       });
 
       it('should not emit events by setting checked programmatically [attribute]', async () => {
-        const changeSpy = await page.spyOnEvent('change');
-        const inputSpy = await page.spyOnEvent('input');
-        const tag2 = await page.find('sbb-tag#sbb-tag-2');
+        const changeSpy = new EventSpy('change');
+        const inputSpy = new EventSpy('input');
+        const tag2 = document.querySelector('sbb-tag#sbb-tag-2') as SbbTag;
 
-        await tag2.removeAttribute('checked');
-        await page.waitForChanges();
+        tag2.removeAttribute('checked');
+        await element.updateComplete;
 
-        expect(tag2).not.toHaveAttribute('checked');
-        expect(await tag2.getProperty('checked')).toBe(false);
-        expect(inputSpy).not.toHaveReceivedEvent();
-        expect(changeSpy).not.toHaveReceivedEvent();
+        expect(tag2).not.to.have.attribute('checked');
+        expect(tag2.checked).to.be.equal(false);
+        expect(inputSpy.count).not.to.be.greaterThan(0);
+        expect(changeSpy.count).not.to.be.greaterThan(0);
       });
     });
 
     describe('initialization', () => {
       it('should read empty array as value [template attribute]', async () => {
-        page = await newE2EPage();
-        await page.setContent(`
-          <sbb-tag-group multiple value='[]'></sbb-tag-group>
-        `);
-        element = await page.find('sbb-tag-group');
-
-        await page.waitForChanges();
+        element = await fixture(html` <sbb-tag-group multiple value="[]"></sbb-tag-group> `);
 
-        expect(await element.getProperty('value')).toEqual([]);
+        await element.updateComplete;
+        expect(element.value).to.be.empty;
```

Comment:
I think empty and empty array is not the same?

Reply:
It should work, but I can add an additional check (`.to.be.an('array').that.is.empty;`) as suggested by the [doc](https://www.chaijs.com/api/bdd/#method_empty:~:text=a%20target%E2%80%99s%20type.-,expect(%5B%5D).to.be.an(%27array%27).that.is.empty%3B,-Add%20.not).